src/mjolnir/landmark_database_builder.cc

```diff
@@ -0,0 +1,273 @@
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include <geos_c.h>
+
+namespace valhalla {
+namespace mjolnir {
+bool LandmarkDatabase::connect_database() {
+  switch (access_mode_) {
+    case AccessMode::ReadWriteCreate:
+      open_flags = SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE;
+      break;
+    case AccessMode::ReadOnly:
+      open_flags = SQLITE_OPEN_READONLY;
+      break;
+    case AccessMode::ReadWrite:
+      open_flags = SQLITE_OPEN_READWRITE;
+      break;
+    default:
+      LOG_ERROR("invalid access mode");
+      return false;
+  }
+
+  if (!LandmarkDatabase::open_database()) { return false; }
+
+  if (access_mode_ == AccessMode::ReadWriteCreate) {
+    if (!create_landmarks_table()) { return false; }
+    if (!create_spatial_index()) { return false; }
+    LOG_INFO("Created database and landmarks table");
+  }
+
+  LOG_INFO("Successfully connected to database");
+  return true;
+}
+
+bool LandmarkDatabase::open_database() {
+  ret = sqlite3_open_v2(database.c_str(), &db, open_flags, NULL);
+  if (ret != SQLITE_OK) {
+    LOG_ERROR("cannot open " + database);
+    sqlite3_close(db);
+    return false;
+  }
+  // loading SpatiaLite as an extension
+  auto db_conn = make_spatialite_cache(db);
+
+  LOG_INFO("Opened database and loaded Spatialite extension");
+  return true;
+}
+
+bool LandmarkDatabase::create_landmarks_table() {
+  sql = "SELECT InitSpatialMetaData(1); CREATE TABLE IF NOT EXISTS landmarks (";
+  sql += "name TEXT,";
+  sql += "type TEXT)";
+
+  ret = sqlite3_exec(db, sql.c_str(), NULL, NULL, &err_msg);
+  if (ret != SQLITE_OK) {
+    LOG_ERROR("Error: " + std::string(err_msg));
+    sqlite3_free(err_msg);
+    sqlite3_close(db);
+    return false;
+  }
+
+  LOG_INFO("Created landmarks table");
+
+  /* creating a POINT Geometry column */
+  sql = "SELECT AddGeometryColumn('landmarks', ";
+  sql += "'geom', 4326, 'POINT', 2)";
+  
+  ret = sqlite3_exec(db, sql.c_str(), NULL, NULL, &err_msg);
+  if (ret != SQLITE_OK) {
+    LOG_ERROR("Error: " + std::string(err_msg));
+    sqlite3_free(err_msg);
+    sqlite3_close(db);
+    return false;
+  }
+
+  LOG_INFO("Added geometry column");
+  return true;
+}
+
+bool LandmarkDatabase::create_spatial_index() {
+  sql = "SELECT CreateSpatialIndex('landmarks', 'geom')";
+  ret = sqlite3_exec(db, sql.c_str(), NULL, NULL, &err_msg);
+  if (ret != SQLITE_OK) {
+    LOG_ERROR("Error: " + std::string(err_msg));
+    sqlite3_free(err_msg);
+    sqlite3_close(db);
+    return false;
+  }
+  LOG_INFO("Created spatial index");
+  return true;
+}
+
+bool LandmarkDatabase::insert_landmark(const std::string& name, const std::string& type, 
+                                      const double longitude, const double latitude) {
+  sql = "INSERT INTO landmarks (name, type, geom) ";
+  sql += "VALUES (?, ?, CastToPoint(MakePoint(?, ?, 4326)))";
```

Comment:
`CastToPoint` seems like a no-op here, `MakePoint` should also create that exact geom. You can try `ST_Point`, which doesn't take the SRID (it's already defined on the column when you create the table)

But yeah, repeating the same thing on the command line is a good way to debug:)